### PR TITLE
docs: create initial changelog

### LINKED ISSUE #770: Add inital changelog

### PR DESCRIPTION
**Pull request by juliapampus**
## What this PR changes/adds:

This PR adds the initial `CHANGELOG.md` to the root directory.

## Why it does that

We need to document implementations to make changes (e.g., features and bug fixes) visible to other developers and users.

## Linked Issue(s)

Closes #770 

## Checklist
Not relevant.

### CODE REVIEW

**Comment by juliapampus** [comment]
Feedback on version tags is welcome as 0.0.1 already exists but does not refer to any milestone.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
+# Changelog
+
+All notable changes to this project will be documented in this file - formatted and maintained 
+according to the rules documented on http://keepachangelog.com.
```
Comment (florianrusch-zf) [inline CHANGELOG.md:4]: I would recommend to put [urls in angle brackets](https://www.markdownguide.org/basic-syntax/#urls-and-email-addresses) for the case, that the rendering engine doesn't support [Automatic URL Linking](https://www.markdownguide.org/extended-syntax/#automatic-url-linking):
```suggestion
according to the rules documented on <http://keepachangelog.com>.
```

**Review by mspiekermann — APPROVED** [review]

**Review by ndr-brt — APPROVED** [review]
Looks good

**Review by paullatzelsperger — APPROVED** [review]
I haven't taken the time to check/cross-ref the issues TBH. However, I'd prefer to not have unreleased stuff in the changelog.

**Review by ndr-brt — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+### Data Plane Framework
+...
+
+### IDS
+
+#### Added
+
+#### Changed
+* Change scope used for obtaining a token for ids multipart (#731)
+* Refactor ids token validation as extension (#625)
```
Comment (paullatzelsperger) [inline CHANGELOG.md:36]: How about something like this:

```markdown
## [0.2.0] - 2021-02-21 (Milestone 2)

### Overview
- DataPlaneFramework (DPF) - first working version
- Data Management API - controller stubs and OpenApi Spec
- Code Quality: Coverage Analysis, System Test Framework
- Traceability and Metrics

### Detailed Changes

#### Added
* `ContractDefinitionStore` supports paging (#717)
* Add okhttp client timeouts (#735)

#### Changed
* Change scope used for obtaining a token for ids multipart (#731)
* Refactor ids token validation as extension (#625)

#### Removed
* Remove ION extension (#664)

#### Fixed

```

Comment (juliapampus) [reply CHANGELOG.md]: @paullatzelsperger Could you pls take another look? I also added some details to the CONTRIBUTING.md.

**Review by paullatzelsperger — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (paullatzelsperger) [reply CHANGELOG.md]: nice, looks good. can be merged IMO.

**Pull request merged**